great_expectations/rule_based_profiler/parameter_builder/numeric_metric_range_multi_batch_parameter_builder.py

```diff
@@ -0,0 +1,262 @@
+        data_context: Optional[DataContext] = None,
+        batch_request: Optional[Union[dict, str]] = None,
+    ):
+        """
+        Args:
+            parameter_name: the name of this parameter -- this is user-specified parameter name (from configuration);
+            it is not the fully-qualified parameter name; a fully-qualified parameter name must start with "$parameter."
+            and may contain one or more subsequent parts (e.g., "$parameter.<my_param_from_config>.<metric_name>").
+            metric_name: the name of a metric used in MetricConfiguration (must be a supported and registered metric)
+            metric_domain_kwargs: used in MetricConfiguration
+            metric_value_kwargs: used in MetricConfiguration
+            false_positive_rate: user-configured fraction between 0 and 1 -- "FP/(FP + TN)" -- where:
+            FP stands for "false positives" and TN stands for "true negatives"; this rate specifies allowed "fall-out"
+            (in addition, a helpful identity used in this method is: false_positive_rate = 1 - true_negative_rate).
+            round_to_nearest_integer: user-configured boolean directive for whether or not to round the computed
+            parameter values (i.e., min_value, max_value) to the nearest integer prior to packaging them on output.
+            data_context: DataContext
+            batch_request: specified in ParameterBuilder configuration to get Batch objects for parameter computation.
+        """
+        super().__init__(
+            parameter_name=parameter_name,
+            data_context=data_context,
+            batch_request=batch_request,
+        )
+
+        self._metric_name = metric_name
+        self._metric_domain_kwargs = metric_domain_kwargs
+        self._metric_value_kwargs = metric_value_kwargs
+
+        self._false_positive_rate = false_positive_rate
+        self._round_to_nearest_integer = round_to_nearest_integer
+
+    def _build_parameters(
+        self,
+        parameter_container: ParameterContainer,
+        domain: Domain,
+        validator: Validator,
+        *,
+        variables: Optional[ParameterContainer] = None,
+        parameters: Optional[Dict[str, ParameterContainer]] = None,
+    ):
+        """
+        Builds ParameterContainer object that holds ParameterNode objects with attribute name-value pairs and optional details.
+            Args:
+        :return: a ParameterContainer object that holds ParameterNode objects with attribute name-value pairs and optional details
+
+        The algorithm operates according to the following steps:
+        1. Obtain batch IDs of interest using DataContext and BatchRequest (unless passed explicitly as argument).  Note
+        that this particular BatchRequest was specified as part of configuration for the present ParameterBuilder class.
+        (This is in contrast to the BatchRequest specified in Checkpoint configuration, or in pipeline, notebook, etc.)
```

Comment:
I'm still confused here. I think we want there to be a single unambiguous batch request, and this doesn't feel like we've boiled out the complexity.

If I say:
`profiler.profile(batch)` then I think we should *only* allow relative keys (e.g. index) in the rule based profiler config.

If I say:
`profiler.profile(data_context=data_context)` then it makes sense to allow an arbitrary (multi) batch request

If I say:
`profiler.profile(batch_request_that_returns_lots_of_batches)` then it makes sense to use a multibatch parameter builder with no other configuration

Pretty much anything else seems like something we should just call an error or at least discourage in docs and warn about.

Reply:
@jcampbell Please see above -- I am changing this to just get the validator from the batch request specified in the parameter builder configuration (and loaded batch IDs from that validator).  This is in a separate branch.

Reply:
I'm a little confused with the first item: if a user passes a single batch via `profiler.profile(batch)`, what would "relative" mean? Would we use the same datasource/connector/asset as that batch to get other batches?